test/cctest/test_aliased_buffer.cc

```diff
@@ -0,0 +1,220 @@
+
+#include "v8.h"
+#include "aliased_buffer.h"
+#include "node_test_fixture.h"
+
+using node::AliasedBuffer;
+
+class AliasBufferTest : public NodeTestFixture {
+ protected:
+  void SetUp() override {
+    NodeTestFixture::SetUp();
+  }
+
+  void TearDown() override {
+    NodeTestFixture::TearDown();
+  }
+};
+
+template<class NativeT>
+void CreateOracleValues(NativeT* buf, size_t count) {
+  for (size_t i = 0, j = count; i < count; i++, j--) {
+    buf[i] = static_cast<NativeT>(j);
+  }
+}
+
+template<class NativeT, class V8T>
+void WriteViaOperator(
+  AliasedBuffer<NativeT, V8T>& aliasedBuffer,  // NOLINT(runtime/references)
+  size_t size,
+  NativeT* oracle) {
+  // write through the API
+  for (size_t i = 0; i < size; i++) {
+    aliasedBuffer[i] = oracle[i];
+  }
+}
+
+template<class NativeT, class V8T>
+void WriteViaSetValue(
+  AliasedBuffer<NativeT, V8T>& aliasedBuffer,  // NOLINT(runtime/references)
+  size_t size, NativeT* oracle) {
+  // write through the API
+  for (size_t i = 0; i < size; i++) {
+    aliasedBuffer.SetValue(i, oracle[i]);
+  }
+}
+
+template<class NativeT, class V8T>
+void ReadAndValidate(
+  v8::Isolate* isolate,
+  v8::Local<v8::Context> context,
+  AliasedBuffer<NativeT, V8T>& aliasedBuffer,  // NOLINT(runtime/references)
+  size_t size,
+  NativeT* oracle) {
+  // read through the API
+  for (size_t i = 0; i < size; i++) {
+    NativeT v1 = (NativeT)aliasedBuffer[i];
+    NativeT v2 = (NativeT)aliasedBuffer.GetValue(i);
```

Comment:
style nit: Can you use `static_cast` for these?

Reply:
removed cast, wasn't necessary.